File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        
        if(numRows == 1){
            List<Integer> col = new ArrayList<>();
            col.add(1);
            res.add(col);
            return res;
        }

        res = generate(numRows-1);
        List<Integer> p = res.get(numRows-2);
        List<Integer> c = new ArrayList<>();
        c.add(1);
        for(int i=1;i<numRows-1;i++){
            c.add(p.get(i-1) + p.get(i));
        }
        c.add(1);
        res.add(c);
        return res;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        row=[[1]]
        for i in range(1,numRows):
            col=[]
            for j in range(0,i+1):
                if j==0 or j==i:
                    col.append(1)
                else:
                    col.append(row[i-1][j]+row[i-1][j-1])
            row.append(col)
        return row